### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by drknzz**
Fix a part of unit tests

### CODE REVIEW

File: starknet_py/compile/compiler.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def starknet_compile(
         main_scope=MAIN_SCOPE,
         add_debug_info=False,
         file_contents_for_debug_info=file_contents_for_debug_info,
-        filter_identifiers=False,
```
Comment (war-in) [inline starknet_py/compile/compiler.py:132]: 👀

File: starknet_py/tests/e2e/client/fixtures/prepare_network.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def fixture_contract_address(prepare_network: Tuple[str, PreparedNetworkData]) -
     return prepared_data.contract_address
 
 
-@pytest.fixture(name="balance_contract", scope="package")
+@pytest.fixture(name="balance_contract", scope="module")
```
Comment (war-in) [inline starknet_py/tests/e2e/client/fixtures/prepare_network.py:141]: Why do we need to change the scope?

Comment (war-in) [reply starknet_py/tests/e2e/client/fixtures/prepare_network.py:141]: Discussed. Can be reverted to `package`

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -12,8 +12,9 @@ jobs:
     name: Download deps, install poetry
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
```
Comment (war-in) [inline .github/workflows/checks.yml:15]: Does it mean that if one test fails the others will continue running?

Comment (drknzz) [reply .github/workflows/checks.yml:15]: [Yes](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast)

File: starknet_py/compile/compiler.py (hunk 1 above)
Comment (drknzz) [reply starknet_py/compile/compiler.py:132]: It has been changed in cairo-lang.

File: .github/workflows/checks.yml (hunk 3 above)
Comment (drknzz) [reply .github/workflows/checks.yml:15]: But I don't think we need the matrix anymore, as only python3.9 is supported

Comment (cptartur) [reply .github/workflows/checks.yml:15]: I wouldn't include it if we are using only `3.9` for now anyways. I'd address changing the checks in the separate issue.

Comment (drknzz) [reply .github/workflows/checks.yml:15]: `fail-fast` only applies to a strategy for *matrix* anyway. If set to true, the same jobs (but for different items from matrix) fail, once any job fails.

**Review by war-in — APPROVED** [review]

**Pull request merged**